### PR TITLE
Add outline-document-search extension

### PR DESCRIPTION
**Pull request by dnlfrst**
## Description

_Outline Document Search_ enables users of Raycast to search for documents in [Outline](https://www.getoutline.com/). The extension supports the [cloud instance and self-managed instances](https://www.getoutline.com/pricing) of Outline. To retrieve search results, _Outline Document Search_ [calls Outline's API](https://www.getoutline.com/developers#tag/Documents/paths/~1documents.search/post).

## Screencast

For the extension to work, a key for Outline's API is necessary. Acquiring such a key is straightforward and [explained in Outline's documentation](https://www.getoutline.com/developers#tag/Documents/paths/~1documents.search/post). Since Raycast extension preferences do not support dynamic fields, Outline instances are configured through a JSON file as:

```json
[
  {
    "name": "Personal",
    "url": "https://app.getoutline.com",
    "apiKey": "ol_api_..."
  },
  {
    "name": "Work",
    "url": "https://outline.work.com",
    "apiKey": "ol_api_..."
  }
]
```

Afterward, one selects an Outline instance to search in:

outline-document-search-instance-selection

and types a query to retrieve the matching documents:

outline-document-search-results

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi 👋

Thanks for your contribution 🔥 

Could you only have the content in extensions/... so I'm able to see what's going on, I'm not sure what this is about 🙂

Raycast17052024-QIcTj48r png@2x

**Comment by dnlfrst** [comment]
> Could you only have the content in extensions/... so I'm able to see what's going on, I'm not sure what this is about 🙂

Sorry for the mishap, should be fixed now! 😓

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Is there a way to gracefully showing an error if you select a wrong format instead of crashing?

outline-document-search 2024-05-17 at 14 23 25

- Wouldn't it be better to do the setup using [Form](https://developers.raycast.com/api-reference/user-interface/form) and saving it in [LocalStorage](https://developers.raycast.com/utilities/react-hooks/uselocalstorage) instead?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by dnlfrst** [comment]
> Wouldn't it be better to do the setup using [Form](https://developers.raycast.com/api-reference/user-interface/form) and saving it in [LocalStorage](https://developers.raycast.com/utilities/react-hooks/uselocalstorage) instead?

Sure! I have refactored the extension to manage the Outline instances this way in favor of using the preferences.

> Is there a way to gracefully showing an error if you select a wrong format instead of crashing?

Since the preference is gone, this is no longer an issue!

---

_I have not yet updated the screenshots in case there is still more feedback to be addressed._

File: extensions/outline-document-search/src/InstancesList.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    );
+  } else {
+    return (
+      <List isLoading={isLoading} navigationTitle="Outline Instances">
+        {instances.map((instance, index) => (
+          <InstanceItem
+            {...{ fetchInstances, instance }}
+            key={index}
+            remove={() => removeInstance(instance)}
+            instancesActions={<InstancesActions shortcut={{ key: "a", modifiers: ["cmd", "shift"] }} title="Add" />}
```
Comment (dnlfrst) [inline extensions/outline-document-search/src/InstancesList.tsx:120]: Is there a sensible way to come up with these shortcuts?

Comment (pernielsentikaer) [reply extensions/outline-document-search/src/InstancesList.tsx:120]: We have some default shortcuts in `Keyboard.Shortcut.Common.`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Could you try to add the svg here locally, then it should change color to fit the environment - otherwise `tintColor: Coler.Primary` shoud maybe also work

outline-document-search 2024-05-27 at 07 17 16

**Comment by dnlfrst** [comment]
> Could you try to add the svg here locally, then it should change color to fit the environment - otherwise `tintColor: Coler.Primary` shoud maybe also work

Adding the SVG to the assets was not sufficient. I still kept it there, though. Ultimately, `tintColor: Coler.PrimaryText` was the solution.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**